protocols/grpc/agent.proto

```diff
@@ -37,9 +37,13 @@ service AgentService {
 	rpc TtyWinResize(TtyWinResizeRequest) returns (google.protobuf.Empty);
 
 	// networking
+	// Add interface, the interface should be removed when failed.
 	rpc AddInterface(AddInterfaceRequest) returns (google.protobuf.Empty);
 	rpc RemoveInterface(RemoveInterfaceRequest) returns (google.protobuf.Empty);
-	rpc UpdateInterface(UpdateInterfaceRequest) returns (google.protobuf.Empty);
+	// Update the interface, the agent should try to roll back the operation
+	// when failed. Return the final effective interface config regardless
+	// failed or not.
+	rpc UpdateInterface(UpdateInterfaceRequest) returns (Interface);
 	rpc AddRoute(RouteRequest) returns (google.protobuf.Empty);
 	rpc RemoveRoute(RouteRequest) returns (google.protobuf.Empty);
```

Comment:
We are lacking a UpdateRoutes API.

Reply:
Since we makes `UpdateInterface` declarative, `UpdateRoutes` looks reasonable now. So yes, I think we can add `UpdateRoutes`. 

It's also fine to put it into a new PR for this

Reply:
According to our previous discussion, you hope  `UpdateRoutes` can updates a group of route rules in one request, right?

Then with this `UpdateRoutes` API, I don't see any need for `AddRoute` and `RemoveRoute`. I'm sorry that I'm still not very clear about your idea

Reply:
Yes. Once we have UpdateRoute we will not need Add and Remove I think. Once we add UpdateRoute this PR LGTM.